src/Arguments/ArgumentsResolver.php

```diff
@@ -103,11 +105,15 @@ public static function parseDefinitionsByMethodSignature(
      *
      * @param ContainerInterface $container
      * @param string $definition
+     * @param bool $isClassExpected treat string as class definition in case of class expectation
      * @return ArgumentInterface
      */
-    private static function parseDefinition(ContainerInterface $container, $definition): ArgumentInterface
-    {
-        if (is_string($definition)) {
+    private static function parseDefinition(
+        ContainerInterface $container,
+        $definition,
+        bool $isClassExpected
+    ): ArgumentInterface {
+        if (is_string($definition) && $isClassExpected) {
```

Comment:
Maybe flip operands because checking boolean value of `$isClssExpected` is the faster operation then `is_string`?

Reply:
Wow, this change is breaking use-case when a type of argument is omitted but definition starts with `@`.

Reply:
You mean definition starts with '$', in this case I strongly suggest to explicitly point expected class instead of point scalar types, which also should be pointed as well.

Reply:
I added tests for aliases

Reply:
PHP does not oblige specifying a type of method argument. In this case, a type of argument is not scalar, it's is the mixed type.

Reply:
I suggest to checking that a type is the scalar instead of class.